packages/gallery/src/components/item/imageItem.js

```diff
@@ -67,58 +95,59 @@ export default class ImageItem extends GalleryComponent {
-        ...imageProps
-      };
-      switch (styleParams.imageLoadingMode) {
-        case LOADING_MODE.BLUR:
-          preload = <img
-            alt=''
-            key={'image_preload_blur-' + id}
-            src={createUrl(URL_SIZES.RESIZED, isSEOMode() ? URL_TYPES.SEO : URL_TYPES.LOW_RES)}
-            style={{ ...restOfDimensions, backgroundSize: '0.3px', backgroundRepeat: 'repeat' }}
-            {...preloadProps}
-          />
-          break;
-        case LOADING_MODE.MAIN_COLOR:
-          preload = <img
-            alt=''
-            key={'image_preload_main_color-' + id}
-            src={createUrl(URL_SIZES.PIXEL, isSEOMode() ? URL_TYPES.SEO : URL_TYPES.LOW_RES)}
-            style={restOfDimensions}
-            {...preloadProps}
-          />
-          break;
-      }
+      const imagesComponents =[]
 
-      const highres = utils.isSSR() ? null : <img
-        key={'image_highres-' + id}
-        className={'gallery-item-visible gallery-item gallery-item-hidden gallery-item-preloaded'}
-        data-hook='gallery-item-image-img'
-        alt={alt ? alt : 'untitled image'}
-        src={createUrl(URL_SIZES.RESIZED, isSEOMode() ? URL_TYPES.SEO : URL_TYPES.HIGH_RES)}
-        loading="lazy"
-        onLoad={({ target }) => {
-          target.style.opacity = '1';
-          setTimeout((() => {
-            try {
-              target.parentElement.querySelector('[data-hook="gallery-item-image-img-preload"]').remove()
-            } catch (e) {
-              //
-            }
-          }), 1000);
-        }}
-        style={restOfDimensions}
-        {...imageProps}
-      />
+      if (!isHighResImageLoaded){
+        let preload = null;
+        const preloadProps = {
+          className: 'gallery-item-visible gallery-item gallery-item-preloaded',
+          key: 'gallery-item-image-img-preload',
+          'data-hook': 'gallery-item-image-img-preload',
+          loading: "lazy",
+          ...imageProps
+        };
+        switch (styleParams.imageLoadingMode) {
+          case LOADING_MODE.BLUR:
+            preload = <img
+              alt=''
+              key={'image_preload_blur-' + id}
+              src={createUrl(URL_SIZES.RESIZED, isSEOMode() ? URL_TYPES.SEO : URL_TYPES.LOW_RES)}
+              style={{ ...restOfDimensions, backgroundSize: '0.3px', backgroundRepeat: 'repeat', transition: 'all 3s ease-in-out' }}
+              {...preloadProps}
+            />
+            break;
+          case LOADING_MODE.MAIN_COLOR:
+            preload = <img
+              alt=''
+              key={'image_preload_main_color-' + id}
+              src={createUrl(URL_SIZES.PIXEL, isSEOMode() ? URL_TYPES.SEO : URL_TYPES.LOW_RES)}
+              style={restOfDimensions}
+              {...preloadProps}
+            />
+            break;
+        }
+
+        imagesComponents.push(preload);
+      }
 
-      return [preload, highres]
+      const shouldRenderHighResImages = !isPrerenderMode() && !utils.isSSR();
```

Comment:
why are we using isSSR?
We should use componentDidMount to allow highres images

Reply:
backward compatibility, we will remove it when Santa will use prerender mode